assets/js/phoenix_live_view.js

```diff
@@ -2924,6 +2925,15 @@ class ViewHook {
   }
 
   __trigger__(kind){
+    if(kind === 'disconnected') {
+      this.__disconnected = true
```

Comment:
We could set this wherever `__trigger__('disconnected')` gets called, but given we deal with private state I think it's better to intercept the event here.